api/v1alpha1/redisshard_types.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+var (
+	redisShardDefaultImage defaultImageSpec = defaultImageSpec{
+		Name:       pointer.StringPtr("redis"),
+		Tag:        pointer.StringPtr("4.0.11-alpine"),
+		PullPolicy: (*corev1.PullPolicy)(pointer.StringPtr(string(corev1.PullIfNotPresent))),
+	}
+	redisShardDefaultMasterIndex int32 = 0
+	RedisShardDefaultReplicas    int32 = 3
+)
+
+// RedisShardSpec defines the desired state of RedisShard
+type RedisShardSpec struct {
+	// Image specification for the component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Image *ImageSpec `json:"image,omitempty"`
+	// MasterIndex is the StatefulSet Pod index of the redis server
+	// with the master role. The other Pods are slaves of the master one.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	MasterIndex *int32 `json:"masterIndex,omitempty"`
```

Comment:
I know RedisShard is only for testing, but trying to understand it

So you need to tell in the CR spec which one will be the master, but it is used only initially? then it is sentinel who manages which one is master/slaves on the CR status?

Reply:
That's right, by default it always configures index 0 as the master. Once the shard has the initial configuration applied it never tries to reconcile the roles of the nodes.

Reply:
ok, thanks for confirming my suspicions :)